### DISCUSSION TITLE
Enum in expression tree

### DISCUSSION THREAD

**Discussion by cheng93** [dotnet/roslyn#65022] [General]
**Version Used**: 

latest

**Steps to Reproduce**:

1. See Sharplab

(A [sharplab](https://sharplab.io/) repro is ideal for compiler/language issues whenever possible)

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6AGQEsoBHMgUQA8AHMAUwGdOqB7KTgG48eVAGYS6IgGEieAN54iyyeiUrFuFdqLM2XHvwA8qAKxGAKl2DSANgENuMIgCNevWwD5PRdqw7cfFBEALxETKE+TIxQEAC2oWFWnMAMsXFkAGLuwlo6ygBu9mCuNMUAnnoBhsFhfvqB/GQAQrxw5USORM1lYJX+BkG5+crq+UUlYFQA5gAWwKGlUBVVg00ASjPzuWM6pACcABRTc8BkFrwAysBTUNOHAJQPwyNUAGZEx1sLVJxEAKrLPqrRrBCAPXbaTQjPaYI4AImgFXhz0hKgAvmjlOxbJx2ER3p8TvMCX9pPwUvYoKkBqCiABjCF5fLQmEqA6HeH0inAKnAFEvfLotkqXaY3Di0QSdjpIjJVLpBS7bK8RaYGC7ZrFRZqCUifASDBy6x2RycJXMkgSeVpeK6WXyIjTdjAQR411EcXooA=

**Expected Behavior**:

I would expect it to be a UnaryExpression, something like `Convert(Foo)`

**Actual Behavior**:

The enum is being prematurely evaluated as a constant.

So when I'm traversing the expression tree, I can't tell the difference whether it is an enum, or it's underlying type

I found some stackoverflow posts, albeit very old relating to this issue

https://stackoverflow.com/questions/34099809/expressions-breaking-code-when-compiled-using-vs2015-update-1
https://stackoverflow.com/questions/34400586/c-sharp-expression-vs-converts-enum-to-int-but-not-linqpad

**Comment by CyrusNajmabadi** [comment]
Enums are constants and their values are embedded into the callsites where they are referenced. So this matches normal c# concepts. This also seems to be what the behavior has been for expression trees for more than 15 years now. So this definitely can't change as it would likely break existing applications.

**Comment by cheng93** [comment]
Actually this would probably be best as a discussion, not sure how to convert it

**Comment by cheng93** [comment]
I have a similar sharplab which does expose the enum

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMA6AGQEsoBHMgUQA8AHMAUwGdOqB7KTgG48eVAGYS6IgGEieAN54iyyeiUrFuFdqLM2XHvwA8qAKxGAKl2DSANgENuMIgCNevWwD5PRdqw7cfFBEALxETKE+TIxQEAC2oWFWnMAMsXFkAGLuwlo6ygBu9mCuNMUAnnoBhsFhfvqB/GQAQrxw5USORM1lYJX+BkG5+crq+UUlYFQA5gAWwKGlUBVVg00ASjPzuWM6pACcABRTc8BkFrwAysBTUNOHAJQPwyNUAGZEx1sLVJxEAKrLPqrRrBCAPXbaTQjPaYI4AImgFXhz0hKgAvmjlOxbJx2ER3p8TvMCX9pPwUvYoKkBqCiABjCF5fLQmEqA6HeH0inAKnAFEvfLotkqXaY3Di0QSdjpIjJVLpBS7bK8RaYGC7ZrFRZqCUifASDBy6x2RycJXMkgSeVpeIAfl0svkRGm7GAgjx7qI4vRQA==

note in this case the property is `TestEnum?`

**Comment by CyrusNajmabadi** [comment]
One is a nullable, the other is a direct reference.  A nullable isn't a constant, so it isn't inlined at the reference sites.
